src/Lean/Language/Lean.lean

```diff
@@ -630,16 +625,24 @@ where
     -- definitely resolve eventually
     snap.new.resolve <| .ofTyped { diagnostics := .empty : SnapshotLeaf }
 
-    let mut infoTree := cmdState.infoState.trees[0]!
+    let mut infoTree : InfoTree := cmdState.infoState.trees[0]!
```

Comment:
Why do need to provide the type annotation here? Is the goal to improve readability or workaround an elaboration issue?

Reply:
The latter, filed as #6237